### ISSUE TITLE
Pod "XYZ" does not exist

### ISSUE THREAD

**Issue by reiz** [kubernetes/kubernetes#17949]
I'm running a Kubernetes 1.1.2 on CoreOS. In general the cluster is working. I can run pods and the applications are responding. If I run `kubectl get pods` I'm getting this output: 

```
NAME            READY     STATUS    RESTARTS   AGE
example-7ubxo   1/1       Running   0          4m
example-ypfk0   1/1       Running   0          4m
```

but `kubectl logs example-7ubxo` returns: 

```
Pod "example-7ubxo" does not exist
```

How can that be?

**Comment by dchen1107** [comment]
cc/ @yifan-gu

**Comment by dchen1107** [comment]
@yifan-gu I marked this as P1 for initial investigation to figure out whether the issue is related to journalctl logs or not. The issue could be a bug in kubelet too. Thanks!

**Comment by reiz** [comment]
@dchen1107 @yifan-gu In the mean I solved this issue. The problem was related to our DNS setup. The minion/coreos nodes are registered in our DNS. Unfortunately we had several IPs to the same DNS entry. After I cleared the DNS and setup a new K8S cluster, all this worked again like expected. 

The error message in this case is misleading. It seems that kubectl sometimes works with IPs and sometimes with DNS entries.
